dokan/loaddll.go

```diff
@@ -0,0 +1,144 @@
+// Copyright 2018 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+// +build windows
+
+package dokan
+
+/*
+#include "bridge.h"
+*/
+import "C"
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"fmt"
+	"io"
+	"os"
+	"runtime"
+	"strings"
+	"unsafe"
+
+	"golang.org/x/sys/windows"
+)
+
+const shortPath = `DOKAN1.DLL`
+const syswow64 = `C:\WINDOWS\SYSWOW64\`
+const system32 = `C:\WINDOWS\SYSTEM32\`
+
+type errorPrinter struct {
+	buf bytes.Buffer
+}
+
+func (ep *errorPrinter) Printf(s string, os ...interface{}) {
+	fmt.Fprintf(&ep.buf, s, os...)
+}
+
+// loadLibrary calls win32 LoadLibrary and logs the result.
+func loadLibrary(epc *errorPrinter, path string) (windows.Handle, error) {
+	hdl, err := windows.LoadLibrary(path)
+	epc.Printf("LoadLibrary(%q) -> %v,%v\n", path, hdl, err)
+	return hdl, err
+}
+
+// doLoadDLL tries to load the dokan DLL from various locations.
+func doLoadDLL(epc *errorPrinter, path string) (windows.Handle, error) {
+	var guessPath bool
+	epc.Printf("loadDokanDLL %q\n", path)
+	if path == "" {
+		path = shortPath
+		guessPath = true
+	} else {
+		path = strings.Replace(path, `/`, `\`, -1)
```

Comment:
Can you use `filepath.FromSlash()` here instead?

Reply:
Will do.